control-plane-pki-operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -90,23 +90,33 @@ func createTargetConfig(ctx context.Context, c TargetConfigController, recorder
 		Type:   "TargetConfigControllerDegraded",
 		Status: metav1.ConditionFalse,
 	}
-	if _, err := clienthelpers.UpdateHostedControlPlaneStatusCondition(ctx, condition, c.namespace, c.name, c.hypershiftClient); err != nil {
+	if _, err := clienthelpers.UpdateHostedControlPlaneStatusCondition(ctx, condition, c.hostedControlPlane.Namespace, c.hostedControlPlane.Name, c.hypershiftClient); err != nil {
```

Comment:
In follow up we'll most likely want to bubble this up to the HC resource and include a check in validateHostedClusterConditions

Reply:
Would you prefer reporting in both spots? Or just HostedCluster instead of HostedControlPlane?

Reply:
for cpo managed components the flow is usually to let them set their condition in the HCP then we bubble them up here https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L533.

I like it how you have it in the PR right now scoped to the HCP, we can later follow up to bubble it up.

Reply:
Sounds good - please let me know what you expect for the roll-up (do we want all N conditions copied up, or an aggregate/summary?) and I'll add in the follow-up.